### ISSUE TITLE
mascli:10.3.1 and 10.4.0 error

### ISSUE THREAD

**Issue by lu4t** [ibm-mas/installer#1127]
when running:

docker run -ti --pull always -v /home:/mascli/masconfig quay.io/ibmmas/cli:latest
latest: Pulling from ibmmas/cli
Digest: sha256:fd57835f6c967fa91194a86fd84b9b252460da9ae641c90968565db472c87139
the Image is 10.3.1

you get this error just before the install pipeline is launched:

21) Launch Install
If you are using storage classes that utilize 'WaitForFirstConsumer' binding mode choose 'No' at the prompt below
Wait for PVCs to bind? [y/n] y
Traceback (most recent call last):
  File "/opt/app-root/bin/mas-cli", line 54, in <module>
    app.install(argv[2:])
  File "/opt/app-root/lib64/python3.9/site-packages/mas/cli/install/app.py", line 941, in install
    installOpenShiftPipelines(self.dynamicClient)
  File "/opt/app-root/lib64/python3.9/site-packages/mas/devops/tekton.py", line 75, in installOpenShiftPipelines
    foundReadyWebhook = waitForDeployment(dynClient, namespace="openshift-pipelines", deploymentName="tekton-pipelines-webhook")
  File "/opt/app-root/lib64/python3.9/site-packages/mas/devops/ocp.py", line 110, in waitForDeployment
    if deployment.status.readyReplicas > 0:
TypeError: '>' not supported between instances of 'NoneType' and 'int'

**Comment by lu4t** [comment]
Same happens is mascli:10.4.0

here is the error before launching install:

```
Traceback (most recent call last):
  File "/opt/app-root/bin/mas-cli", line 54, in <module>
    app.install(argv[2:])
  File "/opt/app-root/lib64/python3.9/site-packages/mas/cli/install/app.py", line 960, in install
    installOpenShiftPipelines(self.dynamicClient)
  File "/opt/app-root/lib64/python3.9/site-packages/mas/devops/tekton.py", line 75, in installOpenShiftPipelines
    foundReadyWebhook = waitForDeployment(dynClient, namespace="openshift-pipelines", deploymentName="tekton-pipelines-webhook")
  File "/opt/app-root/lib64/python3.9/site-packages/mas/devops/ocp.py", line 110, in waitForDeployment
    if deployment.status.readyReplicas > 0:
TypeError: '>' not supported between instances of 'NoneType' and 'int'
```

The problem arises because deployment.status.readyReplicas can be None, and Python cannot compare None with an integer. 
A condition is not None should be added, so that the comparison > can be performed if deployment.status.readyReplicas has a valid integer value. To avoid the TypeError and ensures that the code runs correctly even if deployment.status.readyReplicas is None.

**Comment by lu4t** [comment]
Modifiying the ocp.py script the error is gone. Here are the steps:

from within the mascli container, run:

 `nano /opt/app-root/lib64/python3.9/site-packages/mas/devops/ocp.py`

Add the line: 
`if deployment.status.readyReplicas is not None and deployment.status.readyReplicas > 0:`

comment out the line: 
`if deployment.status.readyReplicas > 0:`

This is how it looks:

```
def waitForDeployment(dynClient: DynamicClient, namespace: str, deploymentName: str) -> bool:
    deploymentAPI = dynClient.resources.get(api_version="apps/v1", kind="Deployment")
    maxRetries = 100
    foundReadyDeployment = False
    retries = 0
    while not foundReadyDeployment and retries < maxRetries:
        retries += 1
        try:
            deployment = deploymentAPI.get(name=deploymentName, namespace=namespace)
   #         if deployment.status.readyReplicas > 0:
            if deployment.status.readyReplicas is not None and deployment.status.readyReplicas > 0:
                foundReadyDeployment = True
            else:
                logger.debug("Waiting 5s for deployment {deploymentName} to be ready before checking again ...")
                sleep(5)
        except NotFoundError:
            logger.debug("Waiting 5s for deployment {deploymentName} to be created before checking again ...")
            sleep(5)
    return foundReadyDeployment
```

**Comment by lu4t** [comment]
However, the script seems to be correct in here: https://github.com/ibm-mas/python-devops/blob/stable/src/mas/devops/ocp.py

I guess something is wrong on the build.

**Comment by durera** [comment]
Looks like we build the CLI too soon after the python package was published and it wasn't picked up ...

```
[ibmmas/cli:10.4.0]mascli$ pip list
Package             Version
------------------- -----------
...
mas-devops          1.3.0
...
```

The fix is in 1.3.1

**Comment by durera** [comment]
Fixed in `10.4.1`

```
[ibmmas/cli:10.4.1]mascli$ pip list | grep mas-
mas-cli             10.4.1
mas-devops          1.3.1
```
